File: core/src/main/java/heroes/journey/entities/tagging/Operation.java
package heroes.journey.entities.tagging;

import java.util.function.BiFunction;

public enum Operation {
    ADD((a, b) -> a + b), SUBTRACT((a, b) -> a - b), MULTIPLY((a, b) -> a * b), DIVIDE(
        (a, b) -> b == 0 ? a : a / b);

    private final BiFunction<Integer, Integer, Integer> operation;

    Operation(BiFunction<Integer, Integer, Integer> op) {
        this.operation = op;
    }

    public int apply(int a, int b) {
        return operation.apply(a, b);
    }
}

File: modlib/src/main/java/heroes/journey/modlib/attributes/Operation.kt
package heroes.journey.modlib.attributes

import java.util.function.BiFunction

enum class Operation(private val operation: BiFunction<Int, Int, Int>) {
  ADD(BiFunction { a: Int, b: Int -> a + b }), SUBTRACT(
    BiFunction { a: Int, b: Int -> a - b }),
  MULTIPLY(
    BiFunction { a: Int, b: Int -> a * b }),
  DIVIDE(
    BiFunction { a: Int, b: Int -> if (b == 0) a else a / b });

  fun apply(a: Int, b: Int): Int {
    return operation.apply(a, b)
  }
}
